<div>
    <h1 style="text-align: center;">ACTIVITY 1 : PALMER PENGUINS DATA ANALYTICS PROJECT</h1>
    <hr>
    <h3 style="text-align: center;">Analyst : Jay Cee I. Macaraig</h3>
</div>

In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<h3>Converting Datasets to DataFrames</h3>
<br>
<h2>Requirements</h2>
<ul>
    <li>DataFrame name</li>
    <li>File Path</li>
</ul>

In [2]:
data = pd.read_csv("datasets\\penguins.csv")

In [3]:
data

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<br>
<h1 style='text-align: center;'>DESCRIPTIVE ANALYTICS</h1>
<br>
<h4>.describe() function is used for checking general analysis of a datasets</h4>

In [4]:
data.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

<h4>.T function is used for rotating the x-axis and y-axis label of a datasets</h4>

In [5]:
data.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h3>Checking for Null Values</h3>
<h4>.isnull() function is used for checking null values for every row of datas</h4>

In [6]:
data.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

<h3>Summarizing Null Values</h3>
<h4>.sum() is a function for summarizing all data inside a dataset</h4>

In [7]:
data.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h3>.value_counts() is a function that checks a total values of a single column</h3>

In [8]:
data['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [9]:
data['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h2>Filtering a Datasets</h2>
<h3>By using conditional operations inside a list declaration, you can filter datas in a dataset</h3>

In [10]:
data[data['species'] == 'Adelie']

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

<h2>Assigning variable Name</h2>
<h3>By assigning variable name, you can use the code wherever and however you want.</h3>

In [11]:
adelie_df = data[data['species'] == 'Adelie']
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [12]:
gentoo_df = data[data['species'] == 'Gentoo']
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [13]:
chinstrap_df = data[data['species'] == 'Chinstrap']
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

<h2>Using two conditionals</h2>
<h3>Using the <b>(&)</b> operator, you can use two conditionals to filter an specific data</h3>

In [14]:
chinstrap_above_4g = data[(data['species'] == 'Chinstrap') & (data['body_mass_g'] >= 4000)]

In [15]:
chinstrap_above_4g

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

<h3>Using .info() to find the number of items in a dataframe. (You can use this in any dataframe)</h3>

In [16]:
chinstrap_above_4g.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


<h2>Changing Data Types</h2>
<h3>using .astype() to assign Data Types like 'category'</h3>

In [17]:
data['species'] = data['species'].astype('category')
data['island'] = data['island'].astype('category')
data['year'] = data['year'].astype('category')
data['sex'] = data['sex'].astype('category')

In [18]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     342 non-null    float64 
 3   bill_depth_mm      342 non-null    float64 
 4   flipper_length_mm  342 non-null    float64 
 5   body_mass_g        342 non-null    float64 
 6   sex                333 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


<br>
<h1 style='text-align: center;'>FILLING UP NULL VALUES</h1>
<br>
<h4>using .fillna() function, you can fill up null values in a certain column</h4>
<br>
<h3>Using Statistical Function</h3>
<h4>using basic mathematical function like .mean(), .median(), .mode() to figure out what data is to replace the null values</h4>

In [19]:
data.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [20]:
data['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [21]:
data['bill_length_mm'] = data['bill_length_mm'].fillna(data['bill_length_mm'].mean())

In [22]:
data.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [23]:
data['bill_depth_mm'].mean()

np.float64(17.151169590643274)

In [24]:
data['bill_depth_mm'] = data['bill_depth_mm'].fillna(data['bill_depth_mm'].mean())

In [25]:
data.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [26]:
data['flipper_length_mm'].mean()

np.float64(200.91520467836258)

In [27]:
data['flipper_length_mm'] = data['flipper_length_mm'].fillna(data['flipper_length_mm'].mean())

In [28]:
data.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [29]:
data['body_mass_g'].mean()

np.float64(4201.754385964912)

In [30]:
data['body_mass_g'] = data['body_mass_g'].fillna(data['body_mass_g'].mean())

In [31]:
data.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [32]:
data['sex'].mode()

0    male
Name: sex, dtype: category
Categories (2, object): ['female', 'male']

In [33]:
data['sex'] = data['sex'].fillna(data['sex'].mode()[0])

In [34]:
data.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<br>
<h3>Adding a custom column in a dataset.</h3>
<h4>Using .columns, you can view the index and column names of a certain dataset.</h4>

In [35]:
data.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [36]:
data['mass_kg'] = np.round(data['body_mass_g'] / 1000, 1)
data

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

<br>
<h1 style='text-align: center;'>DATA AGGREGATION</h1>
<br>
<h4>.groupby() use for Splitting, Apllying, and Compiling data from a dataset</h4>

In [37]:
species_per_island = data.groupby(['species', 'island'])
species_per_island.count()

bill_length_mm  bill_depth_mm  flipper_length_mm  \
species   island                                                        
Adelie    Biscoe                 44             44                 44   
          Dream                  56             56                 56   
          Torgersen              52             52                 52   
Chinstrap Biscoe                  0              0                  0   
          Dream                  68             68                 68   
          Torgersen               0              0                  0   
Gentoo    Biscoe                124            124                124   
          Dream                   0              0                  0   
          Torgersen               0              0                  0   

                     body_mass_g  sex  year  mass_kg  
species   island                                      
Adelie    Biscoe              44   44    44       44  
          Dream               56   56    56       56  
          Torgersen           52   52    52       52  
Chinstrap Biscoe               0    0     0        0  
          Dream               68   68    68       68  
          Torgersen            0    0     0        0  
Gentoo    Biscoe             124  124   124      124  
          Dream                0    0     0        0  
          Torgersen            0    0     0        0

In [38]:
species_per_island = data.groupby(['species', 'island', 'sex'])
species_per_island.size()

species    island     sex   
Adelie     Biscoe     female    22
                      male      22
           Dream      female    27
                      male      29
           Torgersen  female    24
                      male      28
Chinstrap  Biscoe     female     0
                      male       0
           Dream      female    34
                      male      34
           Torgersen  female     0
                      male       0
Gentoo     Biscoe     female    58
                      male      66
           Dream      female     0
                      male       0
           Torgersen  female     0
                      male       0
dtype: int64

<br>
<h2>Correlation v Causation</h2>
<h3>Correlation Analysis</h3>
<hr>

<h4><b>PEARSON R</b> can test correlation between two variables, given that both variables are <b>NUMERICAL DATA</b></h4>
<p>.corr() function is used to find the p-value (probability).</p>
<ul>
    <li> <b>>= 0.5</b> = with correlation</li>
    <li> <b>< 0.5</b> = no correlation</li>
</ul>

In [39]:
flipper_mass = data['body_mass_g'].corr(data['flipper_length_mm'])
flipper_mass

np.float64(0.8712017673060114)

<br>
<p>The probabilty shows that the data is greater than 0.5, with the value of 0.87. <br>Meaning that there is a correlation between the two variable.</p>
<hr>

<h4><b>ONE WAY ANOVA</b> can test correlation between two variables with different data type <b>NUMERICAL DATA -- CATEGORICAL DATA</b></h4>
<p>.f_oneway() is used to find the p_value(probability). this function throws two data, which is <b>stats</b>(not important) and <b>p_value</b>(important)</p>
<ul>
    <li> <b>>= 0.5</b> = with correlation</li>
    <li> <b>< 0.5</b> = no correlation</li>
</ul>

In [40]:
#importing f_oneway from scipy.stats (scince python . stats)
from scipy.stats import f_oneway

In [41]:
data[data['sex'] == 'male']['body_mass_g']
data[data['sex'] == 'female']['body_mass_g']

1      3800.0
2      3250.0
4      3450.0
6      3625.0
12     3200.0
        ...  
335    3525.0
337    3650.0
338    3650.0
340    3400.0
343    3775.0
Name: body_mass_g, Length: 165, dtype: float64

In [104]:
stats, p_value = f_oneway(data[data['sex'] == 'male']['body_mass_g'] , data[data['sex'] == 'female']['body_mass_g'])

p_value

np.float64(3.0226600487050246e-15)

<br>
<p>The probabilty shows that the data is greater than 0.5, with the value of 3.02. <br>Meaning that there is a correlation between the two variable.</p>
<hr>

<br>
<h1 style='text-align: center;'>Exploratory Analysis 1</h1>
<h2>PART 1:</h2>
<br>

<br>
<br>
<h2>QUESTION 1: WHAT SPECIES OF PALMER PENGUIN HAS THE LONGEST FLIPPER?</h2>
<hr>

In [43]:
most_flipper_length = data['flipper_length_mm'].max()
most_flipper_length

np.float64(231.0)

In [44]:
data[data['flipper_length_mm'] == most_flipper_length]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
215  Gentoo  Biscoe            54.3           15.7              231.0   

     body_mass_g   sex  year  mass_kg  
215       5650.0  male  2008      5.6

<br>
<h4>The species of Palmer Penguin that has the longet flipper are <strong>GENTOO</strong></h4>
<p><i>INSIGHT:</i> Shown above is a data of a Palmer Penguin that has the most flipper length in milimeter.</p>
<hr>
<br>

<br>
<br>
<h2>QUESTION 2: WHAT YEAR DOES EACH SPECIES OF PALMER PENGUINS HAS THE HIGHEST BIRTH RATE?</h2>
<hr>

In [45]:
# Using the assigned variable name and .value_counts() function.
adelie_df['year'].value_counts()

year
2009    52
2007    50
2008    50
Name: count, dtype: int64

In [46]:
# Using the assigned variable name and .value_counts() function.
gentoo_df['year'].value_counts()

year
2008    46
2009    44
2007    34
Name: count, dtype: int64

In [47]:
# Using the assigned variable name and .value_counts() function.
chinstrap_df['year'].value_counts()

year
2007    26
2009    24
2008    18
Name: count, dtype: int64

<br>
<h4>The year that has the highest birth rate for a Adelie Palmer Penguins is : <strong>2009</strong></h4>
<h4>While, the year that has the highest birth rate for a Gentoo Palmer Penguins is : <strong>2008</strong></h4>
<h4>Lastly, the year that has the highest birth rate of a Chinstrap Palmer Penguins is : <strong>2007</strong></h4>
<p><i>INSIGHT:</i> Shown above is a data from the year with the highest to lowest birth count or each Palmer Penguin Species.</p>
<hr>
<br>

<br>
<br>
<h2>QUESTION 3: WHAT SEX OF A PALMER PENGUIN HAS THE HIGHEST BILL LENGTH?</h2>
<hr>

In [48]:
highest_bill_length = data['bill_length_mm'].max()
highest_bill_length

np.float64(59.6)

In [49]:
data[data['bill_length_mm'] == highest_bill_length]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
185  Gentoo  Biscoe            59.6           17.0              230.0   

     body_mass_g   sex  year  mass_kg  
185       6050.0  male  2007      6.0

<br>
<h4>The sex of a Palmer Penguin that has the highest bill length is a <strong>male</strong></h4>
<p><i>INSIGHT:</i> From the Gentoo species, this male palmer penguin has the highest bill length recorded.</p>
<hr>
<br>

<br>
<br>
<h2>QUESTION 4: WHAT ISLAND HAS THE MOST POPULATION OF PALMER PENGUINS?</h2>
<hr>

In [50]:
data['island'].values

['Torgersen', 'Torgersen', 'Torgersen', 'Torgersen', 'Torgersen', ..., 'Dream', 'Dream', 'Dream', 'Dream', 'Dream']
Length: 344
Categories (3, object): ['Biscoe', 'Dream', 'Torgersen']

In [51]:
data[data['island'] == 'Biscoe'].info()

<class 'pandas.core.frame.DataFrame'>
Index: 168 entries, 20 to 275
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            168 non-null    category
 1   island             168 non-null    category
 2   bill_length_mm     168 non-null    float64 
 3   bill_depth_mm      168 non-null    float64 
 4   flipper_length_mm  168 non-null    float64 
 5   body_mass_g        168 non-null    float64 
 6   sex                168 non-null    category
 7   year               168 non-null    category
 8   mass_kg            168 non-null    float64 
dtypes: category(4), float64(5)
memory usage: 9.0 KB


In [52]:
data[data['island'] == 'Dream'].info()

<class 'pandas.core.frame.DataFrame'>
Index: 124 entries, 30 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            124 non-null    category
 1   island             124 non-null    category
 2   bill_length_mm     124 non-null    float64 
 3   bill_depth_mm      124 non-null    float64 
 4   flipper_length_mm  124 non-null    float64 
 5   body_mass_g        124 non-null    float64 
 6   sex                124 non-null    category
 7   year               124 non-null    category
 8   mass_kg            124 non-null    float64 
dtypes: category(4), float64(5)
memory usage: 6.8 KB


In [53]:
data[data['island'] == 'Torgersen'].info()

<class 'pandas.core.frame.DataFrame'>
Index: 52 entries, 0 to 131
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            52 non-null     category
 1   island             52 non-null     category
 2   bill_length_mm     52 non-null     float64 
 3   bill_depth_mm      52 non-null     float64 
 4   flipper_length_mm  52 non-null     float64 
 5   body_mass_g        52 non-null     float64 
 6   sex                52 non-null     category
 7   year               52 non-null     category
 8   mass_kg            52 non-null     float64 
dtypes: category(4), float64(5)
memory usage: 3.1 KB


<h4>Shown above is a data of each Island in which it gives the entries of penguins in each island.</h4>
<h1>Or</h1>
<h4>Shown below is a data in which it provide the island that has the most penguins.</h4>

In [54]:
data['island'].describe()

count        344
unique         3
top       Biscoe
freq         168
Name: island, dtype: object

<br>
<h4>The island that has the most population of Palmer Penguins is <strong>Biscoe</strong></h4>
<p><i>INSIGHT:</i> Shown above, Biscoe has the highest entry records that tops the other islands, being 168 population.</p>
<hr>
<br>

<br>
<br>
<h2>QUESTION 5: IN WHICH ISLAND DOES THE PENGUIN WITH THE HIGHEST AVERAGE BILL DEPTH FOR EACH SPECIES RESIDE?</h2>
<hr>

In [55]:
adelie_bill_length = adelie_df['bill_depth_mm'].mean()
gentoo_bill_length = gentoo_df['bill_depth_mm'].mean()
chinstrap_bill_length = chinstrap_df['bill_depth_mm'].mean()
adelie_bill_length

np.float64(18.346357615894043)

In [56]:
adelie_average = adelie_df[adelie_df['bill_depth_mm'] >= adelie_bill_length]
gentoo_average = gentoo_df[gentoo_df['bill_depth_mm'] >= gentoo_bill_length]
chinstrap_average = chinstrap_df[chinstrap_df['bill_depth_mm'] >= chinstrap_bill_length]

In [57]:
adelie_average['island'].value_counts()

island
Dream        29
Torgersen    26
Biscoe       22
Name: count, dtype: int64

In [58]:
gentoo_average['island'].value_counts()

island
Biscoe    63
Name: count, dtype: int64

In [59]:
chinstrap_average['island'].value_counts()

island
Dream    34
Name: count, dtype: int64

<br>
<h4>For Adelie Palmer Penguin, the highest population of penguins with the average bill depth is: <strong>Dream Island</strong></h4>
<h4>While, for Gentoo Palmer Penguin, the highest population of penguins with the average bill depth is no other than: <strong>Biscoe Island</strong></h4>
<h4>Lastly, for Chinstrap Palmer Penguins, the highest population of penguins with the average bill depth is no other than: <strong>Dream Island</strong></h4>
<p><i>INSIGHT:</i> Adelie Palmer Penguins are widely scattered in the different Island, as shown by the data above.</p>
<hr>
<br>

<br>
<br>
<h2>QUESTION 6: HOW MANY FEMALE PALMER PENGUINS DOES EACH ISLAND HAVE?</h2>
<hr>

In [60]:
data[data['island'] == 'Biscoe']['sex'].value_counts()

sex
male      88
female    80
Name: count, dtype: int64

In [61]:
data[data['island'] == 'Dream']['sex'].value_counts()

sex
male      63
female    61
Name: count, dtype: int64

In [62]:
data[data['island'] == 'Torgersen']['sex'].value_counts()

sex
male      28
female    24
Name: count, dtype: int64

<br>
<h4>In Biscoe Island, the population of female Palmer Penguin is: <strong>80</strong></h4>
<h4>While,the population of female Palmer Penguin in Dream Island is: <strong>61</strong></h4>
<h4>Lastly,the population of female Palmer Penguin in Torgersen Island is: <strong>24</strong></h4>
<p><i>INSIGHT:</i> The Biscoe Island would undoubtedly have the highest birth rate, since it has the most female penguins that has the ability to reproduce more offsprings.</p>
<hr>
<br>

<br>
<br>
<h2>QUESTION 7: HOW MANY PALMER PENGUINS HAS BELOW AVERAGE BODY MASS IN 2009?</h2>
<hr>

In [63]:
mass_ave = data[data['year'] == 2009]['mass_kg'].mean()
mass_ave

np.float64(4.2075)

In [64]:
data[(data['mass_kg'] <= mass_ave) & (data['year'] == 2009)].info()

<class 'pandas.core.frame.DataFrame'>
Index: 69 entries, 100 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            69 non-null     category
 1   island             69 non-null     category
 2   bill_length_mm     69 non-null     float64 
 3   bill_depth_mm      69 non-null     float64 
 4   flipper_length_mm  69 non-null     float64 
 5   body_mass_g        69 non-null     float64 
 6   sex                69 non-null     category
 7   year               69 non-null     category
 8   mass_kg            69 non-null     float64 
dtypes: category(4), float64(5)
memory usage: 4.0 KB


<br>
<h4>The number of Penguins that have below average body mass is <strong>69 population</strong></h4>
<p><i>INSIGHT:</i> Shown above is the data that shows the number of population that has below average body mass</p>
<hr>
<br>

<br>
<br>
<h2>PART 2:</h2>

<br>
<br>
<h2>QUESTION 8: WHAT ISLAND HOST THE HIGHEST AVERAGE BODY MASS OF A PALMER PENGUIN PER SPECIES? WHAT VALUE IS THE HIGHEST AVERAGE BODY MASS?</h2>
<hr>

In [98]:
highest_average_mass = np.round(data.groupby(['species', 'island'])['mass_kg'].mean(), 2)
highest_average_mass

species    island   
Adelie     Biscoe       3.72
           Dream        3.69
           Torgersen    3.71
Chinstrap  Biscoe        NaN
           Dream        3.73
           Torgersen     NaN
Gentoo     Biscoe       5.07
           Dream         NaN
           Torgersen     NaN
Name: mass_kg, dtype: float64

<br>
<h4>The island that host the Adelie species with the highest average body mass is : <strong>Biscoe Island</strong></h4>
<h4>While, the Chinstrap species resides in : <strong>Dream Island</strong></h4>
<h4>Lastly, the Gentoo species resides in : <strong>Biscoe Island</strong></h4>
<p><i>INSIGHT:</i> Shown above, the Gentoo species holds the title with the highest average body mass.</p>
<hr>
<br>

   <h3 style="text-align: right; margin-right: 80px;">Analyst : Jay Cee I. Macaraig</h3>
   <h3 style="text-align: right; margin-right: 80px;">BSIT-3B</h3>